### PR TITLE
Add print report notebook

### PR DESCRIPTION
**Pull request by corochann**
`PrintReportNotebook` class is useful when you work with jupyter notebook.
The functionality is originally implemented in [grafi-tt/chaineripy](https://github.com/grafi-tt/chaineripy) by @grafi-tt as chainer Traner's extension.
I modified it to work on current ppe.

Example usage can be found [here](https://www.kaggle.com/corochann/lyft-training-with-multi-mode-confidence).

### CODE REVIEW

File: pytorch_pfn_extras/training/extensions/print_report_notebook.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    to print specified entries of the log in a human-readable format.
+
+    Args:
+        entries (list of str ot None): List of keys of observations to print.
+            If `None` is passed, automatically infer keys from reported dict.
+        log_report (str or LogReport): Log report to accumulate the
+            observations. This is either the name of a LogReport extensions
+            registered to the manager, or a LogReport instance to use
+            internally.
+        out: This is not used, argument is kept to be consistent with `PrintReport`.
```
Comment (corochann) [inline pytorch_pfn_extras/training/extensions/print_report_notebook.py:24]: Again, out is not used but kept to align argument with PrintReport.

**Comment by corochann** [comment]
Review ready @kmaehashi @emcastillo

**Review by emcastillo — APPROVED** [review]

**Pull request merged**